library/src/data/index.js

```diff
@@ -1 +1,25 @@
-window.CoNDeT.data = {}
\ No newline at end of file
+window.CoNDeT.data = {
+  State: (function () {
+    function constructor() {
+      this.state = [];
+      this.subscribers = [];
+    }
+
+    constructor.prototype.subscribe = function (subscriber) {
+      var self = this;
```

Comment:
Tutaj nie jest potrzebne. Chodzi o to, że w JS mamy dynamic this. Cytując z MDN "In most cases, the value of this is determined by how a function is called (runtime binding)". W `callSubscribers` podajesz funkcję anonimową i ona wywoływana jest wewnątrz metody tablicy `forEach`. Z tego co sprawdziłem to `this` powinno tam wskazywać na `window`, ale nie ma ogólnej reguły. Trzeba za każdym razem sprawdzać. Dlatego jak się podaje gdzieś funkcję to robi się trick z `var self = this`. Tutaj jednak nie podajesz callbacku, więc `this` będzie równe `self`